node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -45,30 +44,117 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   implicit def executionContext: ExecutionContext = system.dispatcher
 
-  val peers: Vector[Peer]
-
-  private val _peerServices: mutable.Map[Peer, ServiceIdentifier] =
+  private val _peerData: mutable.Map[Peer, PeerData] =
     mutable.Map.empty
 
-  def peerServices: Map[Peer, ServiceIdentifier] = _peerServices.toMap
+  def peerData: Map[Peer, PeerData] = _peerData.toMap
+
+  def peers: Vector[Peer] = peerData.keys.toVector
+
+  /** Returns peers by querying each dns seed once. These will be IPv4 addresses. */
+  def getPeersFromDnsSeeds: Vector[Peer] = {
+    val dnsSeeds = nodeAppConfig.network.dnsSeeds
+    val addresses = dnsSeeds
+      .flatMap(seed => {
+        try {
+          InetAddress
+            .getAllByName(seed)
+        } catch {
+          case _: UnknownHostException =>
+            logger.debug(s"DNS seed $seed is unavailable")
+            Vector()
+        }
+      })
+      .distinct
+      .filter(_.isReachable(500))
+      .map(_.getHostAddress)
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers.toVector
+  }
 
-  def setPeerServices(
-      peer: Peer,
-      serviceIdentifier: ServiceIdentifier): Unit = {
-    _peerServices.put(peer, serviceIdentifier)
-    ()
+  /** Returns peers from hardcoded addresses taken from https://github.com/bitcoin/bitcoin/blob/master/contrib/seeds/nodes_main.txt */
+  def getPeersFromResources: Vector[Peer] = {
+    val source = Source.fromURL(getClass.getResource("/hardcoded-peers.txt"))
+    val addresses = source
+      .getLines()
+      .toVector
+      .filter(nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  /** Returns all peers stored in db */
+  def getPeersFromDb: Future[Vector[Peer]] = {
+    val addressesF: Future[Vector[PeerDb]] =
+      PeerDAO().findAll()
+    val peersF = addressesF.map { addresses =>
+      val filteredAddresses = addresses.filter(
+        nodeAppConfig.torConf.enabled || _.networkId != AddrV2Message.TOR_V3_NETWORK_BYTE)
```

Comment:
This should probably be turned into a database level query rather than calling `findAll()` which returns everything in table.

Reply:
Looking at `PeerDAO` it might not be easy to do this. Perhaps we should modify the PeerTable to have a tor column?

Reply:
A tor column won't be needed I think. We store the network byte which can be used in database level query.